tests/routes/test_assessment_routes.py

```diff
@@ -1,97 +1,175 @@
+
+    def test_returns_4xx_when_bedrock_returns_mismatched_key_concept(self, mocker, client, lesson_11_request_data, lesson_11_response_body_mismatched):
+        # stub the bedrock response
+        response_body = lesson_11_response_body_mismatched
+        invoke_model_response = {'ResponseMetadata': {'HTTPStatusCode': 200}, 'body': io.StringIO(response_body)}
+        class mock_bedrock_client:
+            def invoke_model(body, modelId, accept, contentType):
+                return invoke_model_response
+        get_bedrock_client_mock = mocker.patch.object(
+            Label,
+            'get_bedrock_client',
+            return_value=mock_bedrock_client
+        )
+
+        # send the flask request
+        request_data = lesson_11_request_data
+        os.environ['AIPROXY_API_KEY'] = 'test_key'
+        response = client.post('/assessment', query_string=request_data, headers={"Content-type": "application/x-www-form-urlencoded", "Authorization": "test_key"})
+
+        assert response.status_code == 400
+
+        # check that get_bedrock_client_mock was called
+        get_bedrock_client_mock.assert_called_once()
+
+
+    def test_should_return_413_on_request_too_large_error(self, mocker, client, randomstring, lesson_11_rubric, bedrock_claude_model, lesson_11_response_body_too_large):
+        # stub the bedrock response
+        response_body = lesson_11_response_body_too_large
+        invoke_model_response = {'ResponseMetadata': {'HTTPStatusCode': 200}, 'body': io.StringIO(response_body)}
+        class mock_bedrock_client:
+            def invoke_model(body, modelId, accept, contentType):
+                return invoke_model_response
+        get_bedrock_client_mock = mocker.patch.object(
+            Label,
+            'get_bedrock_client',
+            return_value=mock_bedrock_client
+        )
+
+        # send the flask request
         os.environ['AIPROXY_API_KEY'] = 'test_key'
         response = client.post('/assessment', query_string={
+          "code": randomstring(10),
           "prompt": randomstring(10),
-          "rubric": randomstring(10),
+          "rubric": lesson_11_rubric,
           "api-key": randomstring(10),
           "examples": "[]",
-          "model": randomstring(10),
+          "model": bedrock_claude_model,
           "remove-comments": "1",
           "num-responses": "1",
           "temperature": "0.2",
         }, headers={"Content-type": "application/x-www-form-urlencoded", "Authorization": "test_key"})
-        assert response.status_code == 400
+        assert response.status_code == 413
 
-    def test_should_return_400_when_no_prompt(self, client, randomstring):
+        get_bedrock_client_mock.assert_called_once()
+
+
+class TestPostAssessmentUnitTests:
+    """ Tests POST to '/assessment' to start an assessment.
+
+    These tests stub out the Label class to test the API endpoint in isolation.
+    """
+
+    def test_should_return_400_when_no_code(self, client, randomstring):
```

Comment:
no change to this test or the ones below, except for `test_should_return_413_on_request_too_large_error` which is moved to above. apologies for the weird diffs.